client/landing/stepper/declarative-flow/internals/components/step-route/index.tsx

```diff
@@ -11,9 +28,92 @@ type StepRouteProps = {
 	renderStep: ( step: StepperStep ) => JSX.Element | null;
 };
 
+//TODO: Check we can move RenderStep function to here and remove the renderStep prop
 const StepRoute = ( { step, flow, showWooLogo, renderStep }: StepRouteProps ) => {
+	const userIsLoggedIn = useSelector( isUserLoggedIn );
+	const loginUrl = useLoginUrlForFlow( { flow } );
 	const stepContent = renderStep( step );
+	const { site, siteSlugOrId } = useSiteData();
 
+	// Ensure that the selected site is fetched, if available. This is used for event tracking purposes.
+	// See https://github.com/Automattic/wp-calypso/pull/82981.
+
+	// TODO: Check if it is duplicated with the useSiteData hook
```

Comment:
Good call -- we should be able to get the `Site` object from `useSiteData()`.

Reply:
I confirmed is possible to use the same site object. 👍